### PR TITLE
Customizable config parser

### PR DESCRIPTION
**Pull request by bolekk**
DecryptionReportingPluginFactory takes an additional config parser that can extract Threshold-specific config from a potentially larger config struct. This will be used by all products embedding Threshold plugin.

### CODE REVIEW

File: go/go.work.sum
Comment (KuphJr) [inline go/go.work.sum:1]: I have not seen a `go.work.sum` file before.  What does this do?

**Review by KuphJr — APPROVED** [review]
Question about a file. Otherwise LGTM!

File: go/go.work.sum
Comment (bolekk) [reply go/go.work.sum:1]: I don't know.

**Review by KuphJr — APPROVED** [review]

**Review by stchrysa — APPROVED** [review]
LGTM

**Pull request merged**